### ISSUE TITLE
loadMessageIndexes() must be called before hasMore()

### ISSUE THREAD

**Issue by fredrikjacobson** [foxglove/mcap#1147]
**Description**

Hi, when trying out mcap using latest version I receive an error that `loadMessageIndexes() must be called before hasMore()` from `ChunkCursor`. When using `2.0.2` the issue seems to disappear. 

- Version:  `@mcap/core": "^2.1.0`
- Platform: Typescript (both node and browser).

**Steps To Reproduce**

Uses FoxGlove nuscenes Mcap file `NuScenes-v1.0-mini-scene-0061-df24c12.mcap`

```typescript
const file = await fetch(
    "https://url-to-mcap/scene.mcap"
  );

const decompressHandlers = await loadDecompressHandlers();
const reader = await McapIndexedReader.Initialize({
  readable: new BlobReadable(await file.blob()),
  decompressHandlers,
});


for await (const message of reader.readMessages({
        topics: ['/CAM_FRONT_RIGHT/image_rect_compressed']
      })) {
    console.log(message.logTime);
}

```

**Expected Behavior**

Should parse message without errors.

**Comment by jtbandes** [comment]
Thanks for the report. I believe this should be fixed by https://github.com/foxglove/mcap/pull/1149.

**Comment by jtbandes** [comment]
Published v2.1.1
